tests/test_home_meter.py

```diff
@@ -123,13 +123,13 @@ def test_event_activate_price_without_market_maker_rate(self):
         self.strategy.bid_update.set_parameters.assert_not_called()
         self.strategy.offer_update.set_parameters.assert_not_called()
 
-    @patch("d3a.models.strategy.home_meter.read_arbitrary_profile")
-    def test_set_energy_forecast_for_future_markets(self, read_arbitrary_profile_mock):
+    @patch("d3a.d3a_core.global_objects_singleton.global_objects.profiles_handler.rotate_profile")
```

Comment:
Are you sure that this shouldn't be the following?
```python
@patch("d3a.models.strategy.home_meter.global_objects.profiles_handler.rotate_profile")
```
I would double-check that the actual patched object is being used. You can read here for more details: [Unittest - Where to patch](https://docs.python.org/3.8/library/unittest.mock.html#id6)

If you find out that my suggestion is correct, it means that tests are probably not doing their job (because they should have failed).

Reply:
(The same applies to the rest of the commit)

Reply:
You are totally right! Will change that.